packages/cdktf/lib/terraform-resource.ts

```diff
@@ -103,7 +108,9 @@ export class TerraformResource
     FileProvisioner | LocalExecProvisioner | RemoteExecProvisioner
   >;
   private _imported?: TerraformResourceImport;
-  private _moved?: TerraformResourceMove;
+  private _movedByTarget?: TerraformResourceMoveByTarget;
```

Comment:
I think one could keep the _moved attribute and combine both types of moves into it with a union.
As you see in the current solution there is now extra book keeping if sth has moved, what type of move it is etc. This is not bad pre-se but can lead to errors due to oversights when next touching this code.

```suggestion
  private _moved?: TerraformResourceMoveByTarget | TerraformResourceMoveById;
```

I would suggest refactoring this in a follow-up PR if there are no strong reasons to keep this design :)

Reply:
This was actually my initial approach for this. Ended up going the way I did because I found that the type checking was quite annoying to work with, and that the different methods of moving required pretty different sets of logic. Though this might just be personal preference 😅. Willing to give it another look in a follow-up PR.